index.js

```diff
@@ -106,12 +115,27 @@ class Logger {
             this.command += CONFIG.SYSTEM.reset;
         }
         console.log(this.command);
+
+        // Writing the previous command to a file if the current command is not a continuation of the previous command
+        if (!stripAnsi(this.command).startsWith(stripAnsi(this.lastCommand))) {
+            this.saveLogToFile()
+        }
         // Save last command if we need to use for joint
         this.lastCommand = this.command;
         this.command = '';
         return this;
     }
 
+    setLogFile(logFilePath) {
```

Comment:
Better to be `setLogStream(stream)`, where `stream` is a [stream object](https://nodejs.org/docs/latest-v5.x/api/stream.html), so that people can determine to pipe the output to stdout, stderr or file stream.

Reply:
let me be more clear.

the use case looks like:

```
logger.setLogStream(process.stdout)
logger.info("hello") // pipe to stdout immediately 
```

```
logger.setLogStream(fs.createWriteStream('example.txt'))
logger.info("hello") // pipe to file immediately 
```

of course, user may not want to create a file stream themselves.
so, probably we can also provide a function `setLogFile` which is based on `setLogStream`

Reply:
I think the latter would be preferable

Reply:
But then another question arises, should `logger.setLogStream(process.stdout)` override the current use of `console.log(***)` in the code or is it only needed as an additional output source? If the latter, it seems strange to me to make additional use of `stdout` or `stderr` when `console.log` is already in use.

Reply:
> I think the latter would be preferable

I agree, normal users may only use the later one, but advanced user may want to define streams themself. Let's support both `setLogStream` and `setLogFile`

> override the current use of console.log(***)?

yeah, it should override. I think we need a wrapper or some better design for handling this.

Reply:
👌